### PR TITLE
Support Content-Type headers with charset/boundary parameters

### PR DESCRIPTION
**Pull request by GocaMaric**
Handler Function Parameters: Changed the function parameters of the returned handler functions to use the http.Handler interface directly. This makes the code more idiomatic and allows for better compatibility with other middleware.

Inline Function Definitions: Eliminated the need for the intermediate fn variable in both SetHeader and AllowContentType functions. You can directly return the http.HandlerFunc in a more concise manner.

Code Comments: Added a comment to explain the splitting of the content type string to extract the MIME type. This improves code readability.

Code Formatting: Applied consistent code formatting to improve readability.

### CODE REVIEW

File: middleware/content_type.go [hunk 1]
```diff
@@ -6,44 +6,40 @@ import (
...
+				// Skip check for empty content body
 				next.ServeHTTP(w, r)
 				return
 			}
 
-			s := strings.ToLower(strings.TrimSpace(r.Header.Get("Content-Type")))
-			if i := strings.Index(s, ";"); i > -1 {
-				s = s[0:i]
-			}
+			s := strings.ToLower(strings.TrimSpace(strings.Split(r.Header.Get("Content-Type"), ";")[0]))
```
Comment (VojtechVitek) [inline middleware/content_type.go:34]: This can panic.

**Review by VojtechVitek — CHANGES_REQUESTED** [review]

File: middleware/content_type.go (hunk 1 above)
Comment (VojtechVitek) [reply middleware/content_type.go:34]: https://go.dev/play/p/ISGDkoiv4Us

**Review by VojtechVitek — APPROVED** [review]
LGTM. Thanks for your contribution :)

File: middleware/content_type.go (hunk 1 above)
Comment (VojtechVitek) [inline middleware/content_type.go:34]: This doesn't change the behavior (much).

The code is slightly cleaner and can trim whitespaces within first part of the header value.

I'll merge.

Comment (GocaMaric) [reply middleware/content_type.go:34]: Thanks for the feedback! I agree that trimming whitespaces and cleaning up the code makes it more efficient and readable. Appreciate you merging the changes!

**Pull request merged**